docs/concepts/what-is-a-signal.md

```diff
@@ -2,38 +2,29 @@
 tags:
   - signals
   - explanation
 ---
 
-A Signal is an external asynchronous request to a [Workflow Execution](/concepts/what-is-a-workflow-execution).
+A Signal is an asynchronous request to a [Workflow Execution](/concepts/what-is-a-workflow-execution).
 
-A Signal is meant to deliver data to a running Workflow Execution which can be used to change variable values and the state of Workflow Execution.
-A Signal can not return data to the caller, use [Queries](/concepts/what-is-a-query) for that.
-A Signal can be sent using a Temporal Client or from within a Workflow.
-When a Signal is sent, it is received by the Cluster and recorded as an Event to the Workflow Execution Event History.
-The Cluster will deduplicate Signals and use the first Signal with a particular Id.
-The next scheduled Workflow Task contains the Signal Event.
+A Signal delivers data to a running Workflow Execution.
+It cannot return data to the caller (use [Queries](/concepts/what-is-a-query) for that).
+The Workflow code that handles a Signal may mutate Workflow state.
+A Signal can be sent from a Temporal Client or a Workflow.
+When a Signal is sent, it is received by the Cluster and recorded as an Event to the Workflow Execution [Event History](/concepts/what-is-an-event-history).
+A successful response from the Cluster means that the Signal has been persisted and will be delivered at least once to the Workflow Execution.[^1]
+The next scheduled Workflow Task will contain the Signal Event.
 
-A Signal is a message with a unique Id.
-A Signal must include a destination (Namespace + Workflow Id).
+A Signal must include a destination (Namespace & Workflow Id) and name.
+It may include a list of arguments.
 
-A Signal Header includes the following:
-
-- Recipient: Workflow Execution (Namespace + Workflow Id)
-- Id: The unique Id of the Signal.
-- Name: The queue in which the Signal will be added.
-
-A Signal Body includes the following:
-
-- Any encodable data.
-
-Workflow functions listen for Signals by the Signal name.
-Signals are delivered in the order they are received.
-Workflow Execution can optionally await on a single Signal name or multiple Signal names.
-
-If you are using Signals with the Go SDK, you should make sure to do an asynchronous drain on the Signal channel or the Signals will be lost.
+Signal handlers are Workflow functions that listen for Signals by the Signal name.
+Signals are delivered in the order they are received by the Cluster.
+If multiple deliveries of a Signal would be a problem for your Workflow, add idempotency logic to your Signal handler that checks for duplicates.
 
 - [How to use Signals](/application-development-guide/#signals)
+
+[^1]: The Cluster usually deduplicates Signals, but does not guarantee deduplication: there can be two Signal Events (and therefore two deliveries to the Workflow Execution) for a single Signal during shard migration, since the deduping info is only stored in memory.
```

Comment:
```suggestion
[^1]: The Cluster usually deduplicates Signals, but does not guarantee deduplication: During shard migration, two Signal Events (and therefore two deliveries to the Workflow Execution) can occur for a single Signal because the deduping info is stored only in memory.
```